sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

```diff
@@ -34,7 +34,7 @@ import org.apache.spark.scheduler.SparkListenerEvent
  * @since 2.0.0
  */
 @Evolving
-abstract class StreamingQueryListener {
+abstract class StreamingQueryListener extends Serializable {
```

Comment:
Serializing the entire class requires to update the `StreamingQueryListener` on server side to extend `Serializable `

Reply:
Client side class is different from server side, right?